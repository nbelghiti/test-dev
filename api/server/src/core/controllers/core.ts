import {
	ApiDefinition,
	ApiOperationGet,
	ApiPath,
	SwaggerModelType,
	SwaggerResponseType,
} from '@studiohyperdrive/swagger-express-ts';

import { version } from '@pkg';
import { NotFoundError } from '@shared/helpers/error';
import { IRequest, IResponse, INext } from '@shared/shared.types';

import { CoreModels } from '../core.models';
import { IStatus } from '../core.types';

@ApiPath({
	name: 'Core',
	path: '/',
})
export class CoreController {
	@ApiDefinition({
		description: 'Error',
		type: SwaggerModelType.OBJECT,
	})
	public static Error = CoreModels.Error;

	@ApiDefinition({
		description: 'ErrorDetail',
		type: SwaggerModelType.OBJECT,
	})
	public static ErrorDetail = CoreModels.ErrorDetail;

	@ApiDefinition({
		description: 'Status',
		type: SwaggerModelType.OBJECT,
	})
	public static Status = CoreModels.Status;

	@ApiDefinition({
		description: 'Auth0Users',
		type: SwaggerModelType.OBJECT,
	})
	public static Auth0Users = CoreModels.Auth0Users;

	@ApiOperationGet({
		description: 'Get status',
		path: '/status',
		responses: {
			200: {
				description: 'OK',
				type: SwaggerResponseType.OBJECT,
				schema: 'Status',
			},
		},
	})
	public status(req: IRequest, res: IResponse, next: INext): IResponse | void {
		return res.status(200).json({
			version,
			success: true,
		} as IStatus);
	}

	// Get fallback
	public fallback(req: IRequest, res: IResponse, next: INext): IResponse | void {
		return next(new NotFoundError());
	}
}
