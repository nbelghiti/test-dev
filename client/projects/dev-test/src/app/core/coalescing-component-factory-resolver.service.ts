/* tslint:disable */
/**
 * See https://github.com/jonrimmer/angular-coalescing-component-factory-resolver/blob/master/src/app/coalescing-component-factory-resolver.service.ts
 * Discussion & bug ticket: https://github.com/angular/angular/issues/14324#issuecomment-481898762
 *
 * Entry components are only available within a lazy loaded module, breaking dynamic generation (e.g. dialogs). This monkey patch enables resolving entry components from lazy loaded modules as well.
 *
 *
 * Provide in root module:
 *
 * @NgModule({
 *   providers: [CoalescingComponentFactoryResolver]
 * })
 * class AppModule {
 *   constructor(coalescingResolver: CoalescingComponentFactoryResolver) {
 *     coalescingResolver.init();
 *   }
 * }
 *
 *
 * Use in lazy loaded module:
 *
 *
 * @NgModule({})
 * export class LazyModule {
 *   constructor(
 *     coalescingResolver: CoalescingComponentFactoryResolver,
 *     localResolver: ComponentFactoryResolver
 *   ) {
 *     coalescingResolver.registerResolver(localResolver);
 *   }
 * }
 *
 *
 * How it works: When initialised, it injects the root app's ComponentFactoryResolver and monkey patches the resolveComponentFactory method
 * to call its own implementation. This implementation first tries resolving the component factory on all the registered lazy-module resolvers,
 * then falls back to the root app resolver (there's a bit of extra logic to avoid cyclic calls, but that's the gist).
 */
/* tslint:enable */

import {
	ComponentFactory,
	ComponentFactoryResolver,
	Injectable,
	Type
} from '@angular/core';

@Injectable()
export class CoalescingComponentFactoryResolver extends ComponentFactoryResolver {
	private rootResolve: (component: Type<any>) => ComponentFactory<any>;

	private inCall = false;

	private readonly resolvers = new Map<
		ComponentFactoryResolver,
		(component: Type<any>) => ComponentFactory<any>
	>();

	constructor(private readonly rootResolver: ComponentFactoryResolver) {
		super();
	}

	init() {
		this.rootResolve = this.rootResolver.resolveComponentFactory;
		this.rootResolver.resolveComponentFactory = this.resolveComponentFactory;
	}

	public registerResolver(resolver: ComponentFactoryResolver) {
		const original = resolver.resolveComponentFactory;
		this.resolvers.set(resolver, original);
	}

	public resolveComponentFactory = <T>(
		component: Type<T>
	): ComponentFactory<T> => {
		// Prevents cyclic calls.
		if (this.inCall) {
			return null;
		}

		this.inCall = true;
		try {
			const result = this.resolveInternal(component);
			return result;
		} finally {
			this.inCall = false;
		}
	}

	private resolveInternal = <T>(component: Type<T>): ComponentFactory<T> => {
		for (const [resolver, fn] of Array.from(this.resolvers.entries())) {
			try {
				const factory = fn.call(resolver, component);
				if (factory) {
					return factory;
				}
			} catch { }
		}

		return this.rootResolve.call(this.rootResolver, component);
	}
}
