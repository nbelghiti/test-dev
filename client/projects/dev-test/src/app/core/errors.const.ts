export const ToastrConfig = {
	maxOpened: 2,
	timeOut: 8000
};
