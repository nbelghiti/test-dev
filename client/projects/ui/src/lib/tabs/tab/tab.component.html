<div
	class="c-tab"
	[hidden]="!active"
>
	<ng-content></ng-content>
</div>