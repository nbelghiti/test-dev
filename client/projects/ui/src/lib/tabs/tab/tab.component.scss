// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Tab
// =============================================================================
.c-tab {
	padding-top: 2rem;
}

// =============================================================================
// :: Elements
// =============================================================================
