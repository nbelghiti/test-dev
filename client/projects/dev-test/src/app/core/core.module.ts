import { ErrorHandler, NgModule } from '@angular/core';

import { CoalescingComponentFactoryResolver } from './coalescing-component-factory-resolver.service';
import { ContextModule } from './context/context.module';
import { GlobalErrorHandler } from './global-error-handler.service';
import { RouterModule } from './router/router.module';
import { UserModule } from './user/user.module';

@NgModule({
	imports: [
		ContextModule,
		RouterModule,
		UserModule,
	],
	providers: [
		CoalescingComponentFactoryResolver,
		{
			provide: ErrorHandler,
			useClass: GlobalErrorHandler,
		},
	],
})
export class CoreModule {
	constructor(coalescingResolver: CoalescingComponentFactoryResolver) {
		coalescingResolver.init();
	}
}
