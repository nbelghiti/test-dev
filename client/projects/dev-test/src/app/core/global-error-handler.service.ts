import { ErrorHandler, Inject, Injectable, Injector } from '@angular/core';
import { TranslateService } from '@ngx-translate/core';
import { ToastrService } from 'ngx-toastr';

@Injectable()
export class GlobalErrorHandler extends ErrorHandler {
	constructor(
		@Inject(Injector) private readonly injector: Injector,
		private translate: TranslateService
	) {
		super();
	}

	handleError(error) {
		this.translate.get('ERRORS__GENERAL').subscribe((message: string) => {
			this.toastr.error(message, null, { onActivateTick: true });
		});

		console.error(error);

		super.handleError(error);
	}

	/**
	 * Need to get ToastrService from injector rather than constructor injection to avoid cyclic dependency error
	 * https://github.com/scttcper/ngx-toastr/issues/327
	 */
	private get toastr(): ToastrService {
		return this.injector.get(ToastrService);
	}
}
