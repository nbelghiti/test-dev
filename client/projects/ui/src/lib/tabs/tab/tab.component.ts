import { Component, Input } from '@angular/core';

@Component({
	selector: 'ui-tab',
	templateUrl: './tab.component.html',
	styleUrls: ['./tab.component.scss'],
})

export class TabComponent {
	@Input() public title: string;
	@Input() public active: boolean = false;
	@Input() public count: number;
	@Input() public link: string | unknown[];
}
