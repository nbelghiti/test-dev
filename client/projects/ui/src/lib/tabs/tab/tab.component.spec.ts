import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { TabComponent } from './tab.component';

describe('TabComponent', () => {
	let component: TabComponent;
	let fixture: ComponentFixture<TabComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
		declarations: [
			TabComponent,
		]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(TabComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
