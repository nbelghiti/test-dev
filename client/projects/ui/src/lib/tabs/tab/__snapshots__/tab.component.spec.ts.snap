// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabComponent Should render correctly 1`] = `
<ui-tab
  active="false"
>
  <div
    class="c-tab"
    hidden=""
  />
</ui-tab>
`;
